contracts/JBBuybackDelegate.sol

```diff
@@ -1,453 +1,540 @@
-        } else {
-            _swapAmountOut = _getQuote(_data.amount.value);
+            // Unpack the quote from the pool, given by the frontend.
+            (_quoteExists, _metadata) = JBDelegateMetadataLib.getMetadata(DELEGATE_ID, _data.metadata);
+            if (_quoteExists) (_minimumSwapAmountOut, _amountToSwapWith) = abi.decode(_metadata, (uint256, uint256));
         }
 
-        // If the minimum amount received from swapping is greather than received when minting, use the swap pathway
-        if (_tokenCount < _swapAmountOut) {
-            // Return this delegate as the one to use, along the quote and reserved rate, and do not mint from the terminal
+        // If no amount was specified to swap with, default to the full amount of the payment.
+        if (_amountToSwapWith == 0) _amountToSwapWith = _data.amount.value;
+
+        // Find the default total number of tokens to mint as if no Buyback Delegate were installed, as a fixed point number with 18 decimals
+        uint256 _tokenCountWithoutDelegate = mulDiv(_amountToSwapWith, _data.weight, _data.amount.decimals);
+
+        // Keep a reference to the project's token.
+        address _projectToken = projectTokenOf[_data.projectId];
+
+        // Keep a reference to the token being used by the terminal that is calling this delegate. Use weth is ETH.
+        address _terminalToken = _data.amount.token == JBTokens.ETH ? address(WETH) : _data.amount.token;
+
+        // If a minimum amount of tokens to swap for wasn't specified, resolve a value as good as possible using a TWAP.
+        if (_minimumSwapAmountOut == 0) {
+            _minimumSwapAmountOut = _getQuote(_data.projectId, _projectToken, _amountToSwapWith, _terminalToken);
+        }
+
+        // If the minimum amount received from swapping is greather than received when minting, use the swap path.
+        if (_tokenCountWithoutDelegate < _minimumSwapAmountOut) {
+            // Make sure the amount to swap with is at most the full amount being paid.
+            if (_amountToSwapWith > _data.amount.value) revert JuiceBuyback_InsufficientPayAmount();
+
+            // Keep a reference to a flag indicating if the pool will reference the project token as the first in the pair.
+            bool _projectTokenIs0 = address(_projectToken) < _terminalToken;
+
+            // Return this delegate as the one to use, while forwarding the amount to swap with. Speficy metadata that allows the swap to be executed.
             delegateAllocations = new JBPayDelegateAllocation3_1_1[](1);
             delegateAllocations[0] = JBPayDelegateAllocation3_1_1({
                 delegate: IJBPayDelegate3_1_1(this),
-                amount: _data.amount.value,
-                metadata: abi.encode(_tokenCount, _swapAmountOut)
+                amount: _amountToSwapWith,
+                metadata: abi.encode(_quoteExists, _projectTokenIs0, _minimumSwapAmountOut, _data.amount.value - _amountToSwapWith, _data.weight)
```

Comment:
If _amountToSwapWith is low (uint256(1)) and we send 1 ETH, _amountToMintWith becomes 0.999~ ETH

Which in didPay() is then added to the total mintTokensOf()

Meaning that the recipient will be minted: ~1 ETH (worth of JBX, swapped from pool) + Remainder left from swap (if any) + _amountToMintWith

Looks like a double spend to me. See: https://github.com/jbx-protocol/juice-buyback/blob/8dd295937a70fb037b4da60c91d422257ff5b1a2/contracts/JBBuybackDelegate.sol#L299-L309